src/liballoc/boxed.rs

```diff
@@ -479,6 +482,15 @@ impl From<Box<str>> for Box<[u8]> {
     }
 }
 
+#[allow(incoherent_fundamental_impls)]
```

Comment:
Please add a comment explaining the implications of this.

Reply:
I don't know what they are.  I'm happy to push comments if anyone has any suggestions!

Reply:
`incoherent_fundamental_impls` is a `C-future-compatibility` warning; those are essentially hard errors that have been temporarily lowered into warnings to give people time to migrate. Tho in this case it doesn't seem clearly specified in the issue what the warning actually does. However, there's a PR up that is actually making it into a hard error https://github.com/rust-lang/rust/pull/49799. If this impl is admitted into stable Rust (i.e. through `#[allow(incoherent_fundamental_impls)]`) then the warning cannot, AFAIK, be made into a hard error. That seems... bad.

cc @nikomatsakis @rust-lang/wg-traits